pkg/cluster/k8sres.go

```diff
@@ -1000,20 +1010,31 @@ func (c *Cluster) getPodEnvironmentSecretVariables() ([]v1.EnvVar, error) {
 	}
 
 	for k := range secret.Data {
-		secretPodEnvVarsList = append(secretPodEnvVarsList,
-			v1.EnvVar{Name: k, ValueFrom: &v1.EnvVarSource{
-				SecretKeyRef: &v1.SecretKeySelector{
-					LocalObjectReference: v1.LocalObjectReference{
-						Name: c.OpConfig.PodEnvironmentSecret,
+		if !isEnvVarPresent(secretPodEnvVarsList, k) {
```

Comment:
Same here. Let's do this in `generateStatefulSet`

Reply:
Fix logic, please review.